plugins/builds/index.js

```diff
@@ -158,6 +180,136 @@ function handleNextBuild({ buildConfig, joinList, finishedBuilds, jobId }) {
+
+    return eventFactory.create(payload);
+}
+
+/**
+ * Create external build
+ * @method createExternalBuild
+ * @param  {Object}   config                    Configuration object
+ * @param  {Factory}  config.pipelineFactory    Pipeline Factory
+ * @param  {Factory}  config.eventFactory       Event Factory
+ * @param  {String}   config.externalPipelineId External pipelineId
+ * @param  {String}   config.externalJobName    External jobName
+ * @param  {Number}   config.parentBuildId      Parent Build Id
+ * @param  {Object}   config.parentBuilds       Builds that triggered this build
+ * @param  {String}   config.causeMessage       Cause message of this event
+ * @param  {Boolean}  [config.start]            Whether to start the build after creating
+ * @return {Promise}
+ */
+async function createExternalBuild(config) {
+    const { pipelineFactory, eventFactory, externalPipelineId, externalJobName,
+        parentBuildId, parentBuilds, causeMessage, start } = config;
+
+    return createEvent({
+        pipelineFactory,
+        eventFactory,
+        pipelineId: externalPipelineId,
+        startFrom: externalJobName,
+        parentBuildId, // current build
+        causeMessage,
+        parentBuilds,
+        start: start !== false
+    });
+}
+
+/**
+ * Create internal build. If config.start is false or not passed in then do not start the job
+ * @method createInternalBuild
+ * @param  {Object}   config                Configuration object
+ * @param  {Factory}  config.jobFactory     Job Factory
+ * @param  {Factory}  config.buildFactory   Build Factory
+ * @param  {Factory}  config.eventFactory   Event Factory
+ * @param  {Number}   config.pipelineId     Pipeline Id
+ * @param  {String}   config.jobName        Job name
+ * @param  {String}   config.username       Username of build
+ * @param  {String}   config.scmContext     SCM context
+ * @param  {Build}    config.build          Build object
+ * @param  {Object}   config.parentBuilds   Builds that triggered this build
+ * @param  {Boolean}  [config.start]        Whether to start the build or not
+ * @param  {String}   config.baseBranch     Branch name
+ * @return {Promise}
+ */
+async function createInternalBuild(config) {
+    const { jobFactory, buildFactory, eventFactory, pipelineId, jobName,
+        username, scmContext, build, parentBuilds, start, baseBranch } = config;
+    const event = await eventFactory.get(build.eventId);
+    const job = await jobFactory.get({
+        name: jobName,
+        pipelineId
+    });
+    const prRef = event.pr.ref ? event.pr.ref : '';
+
+    if (job.state === 'ENABLED') {
+        return buildFactory.create({
+            jobId: job.id,
+            sha: build.sha,
+            parentBuildId: build.id,
+            parentBuilds,
+            eventId: build.eventId,
+            username,
+            configPipelineSha: event.configPipelineSha,
+            scmContext,
+            prRef,
+            start: start !== false,
```

Comment:
```suggestion
            start
```

`start !== false` is equivalent to  `start`. How about to use above style?

Reply:
looks like it doesn't work if `start` is `undefined`
left it as is for now

Reply:
I see